lib/internal/http2/compat.js

```diff
@@ -330,45 +306,94 @@ class Http2ServerResponse extends Stream {
     code |= 0;
     if (code >= 100 && code < 200)
       throw new RangeError('Informational status codes cannot be used');
-    if (code < 200 || code > 999)
+    if (code < 200 || code >= 600)
```

Comment:
The `999` here is intentional for compatibility with the existing core implementation.

https://github.com/nodejs/node/blob/master/lib/_http_server.js#L187